src/util/capabilities.c

```diff
@@ -0,0 +1,208 @@
+/*
+   SSSD
+
+   Capabilities management helpers
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "config.h"
+#include <sys/prctl.h>
+#include <linux/securebits.h>
+#include <sys/capability.h>
+
+#include "util/util.h"
+
+
+typedef struct _cap_description
+{
+    cap_value_t val;
+    const char *name;
+} cap_description;
+
+#define _CAP_DESCR(cap) {cap, #cap}
+
+static cap_description _all_caps[] =
+{
+    _CAP_DESCR(CAP_AUDIT_CONTROL),
+    _CAP_DESCR(CAP_AUDIT_READ),
+    _CAP_DESCR(CAP_AUDIT_WRITE),
+    _CAP_DESCR(CAP_BLOCK_SUSPEND),
+    _CAP_DESCR(CAP_BPF),
+    _CAP_DESCR(CAP_CHECKPOINT_RESTORE),
+    _CAP_DESCR(CAP_CHOWN),
+    _CAP_DESCR(CAP_DAC_OVERRIDE),
+    _CAP_DESCR(CAP_DAC_READ_SEARCH),
+    _CAP_DESCR(CAP_FOWNER),
+    _CAP_DESCR(CAP_FSETID),
+    _CAP_DESCR(CAP_IPC_LOCK),
+    _CAP_DESCR(CAP_IPC_OWNER),
+    _CAP_DESCR(CAP_KILL),
+    _CAP_DESCR(CAP_LEASE),
+    _CAP_DESCR(CAP_LINUX_IMMUTABLE),
+    _CAP_DESCR(CAP_MAC_ADMIN),
+    _CAP_DESCR(CAP_MAC_OVERRIDE),
+    _CAP_DESCR(CAP_MKNOD),
+    _CAP_DESCR(CAP_NET_ADMIN),
+    _CAP_DESCR(CAP_NET_BIND_SERVICE),
+    _CAP_DESCR(CAP_NET_BROADCAST),
+    _CAP_DESCR(CAP_NET_RAW),
+    _CAP_DESCR(CAP_PERFMON),
+    _CAP_DESCR(CAP_SETGID),
+    _CAP_DESCR(CAP_SETFCAP),
+    _CAP_DESCR(CAP_SETPCAP),
+    _CAP_DESCR(CAP_SETUID),
+    _CAP_DESCR(CAP_SYS_ADMIN),
+    _CAP_DESCR(CAP_SYS_BOOT),
+    _CAP_DESCR(CAP_SYS_CHROOT),
+    _CAP_DESCR(CAP_SYS_MODULE),
+    _CAP_DESCR(CAP_SYS_NICE),
+    _CAP_DESCR(CAP_SYS_PACCT),
+    _CAP_DESCR(CAP_SYS_PTRACE),
+    _CAP_DESCR(CAP_SYS_RAWIO),
+    _CAP_DESCR(CAP_SYS_RESOURCE),
+    _CAP_DESCR(CAP_SYS_TIME),
+    _CAP_DESCR(CAP_SYS_TTY_CONFIG),
+    _CAP_DESCR(CAP_SYSLOG),
+    _CAP_DESCR(CAP_WAKE_ALARM)
+};
+
+static inline const char *cap_flag_to_str(cap_flag_value_t flag)
+{
+    if (flag == CAP_SET) {
+        return "*1*";
```

Comment:
Just out of curiosity, is there a reason for wrapping the `1` in `*`?

Reply:
Just to make it more visible during debugging, before I added an option to print only-non-zero caps.

I can remove entire helper (`cap_flag_to_str()`)

Reply:
> Just to make it more visible during debugging, before I added an option to print only-non-zero caps.
> 
> I can remove entire helper (`cap_flag_to_str()`)

No, please keep it.